platform/openide.dialogs/src/org/openide/DialogDisplayer.java

```diff
@@ -133,13 +133,19 @@ public Dialog createDialog(DialogDescriptor descriptor, Frame parent) {
      * @see "#30031"
      */
     private static final class Trivial extends DialogDisplayer {
+        @Override
         public Object notify(NotifyDescriptor nd) {
+            if (GraphicsEnvironment.isHeadless()) {
+                return NotifyDescriptor.CLOSED_OPTION;
```

Comment:
Showing dialog would yield an exception in headless mode. Returning `CLOSED_OPTION` is the best this default, dummy implementation can do, in my opinion.